### ISSUE TITLE
Imposible to use raw data in AbstractRestfulController in some cases

### ISSUE THREAD

**Issue by svycka** [zendframework/zend-mvc#42]
here https://github.com/zendframework/zend-mvc/blob/8a464ff52049a6ad4990347eb30f69fb79f14a47/src/Controller/AbstractRestfulController.php#L586 is unreachable code

```
parse_str($content, $parsedParams); // returned $parsedParams keys not indexes
```

there are only one way to get raw response if $content is something like `0=something` otherwise if `$content = 'foo'` or similar parse_str() will return `['foo' => '']`

**Comment by svycka** [comment]
for example if you use:

```
public function update($name, $data) {
    // here $data is always array but sometimes I need just a string or xml or something.. I want raw data
}
```

**Comment by svycka** [comment]
ping @weierophinney or someone any response?

**Comment by weierophinney** [comment]
Fixed with #43
